django/core/serializers/base.py

```diff
@@ -207,6 +208,49 @@ def save(self, save_m2m=True, using=None, **kwargs):
         # the m2m data twice.
         self.m2m_data = None
 
+    def save_deferred_fields(self, using=None):
+        if self.deferred_fields is None:
+            return
+
+        self.m2m_data = {}
+
+        for field, field_value in self.deferred_fields.items():
```

Comment:
Isn't this block largely duplicated code? Shouldn't it be possible to factor out the "set the value of this m2m/m2o field" logic into a common location so that there's no duplication between first and second pass updates?